cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -544,8 +544,10 @@ func ValidateMixedArguments(flag *pflag.FlagSet) error {
 
 	mixedInvalidFlags := []string{}
 	flag.Visit(func(f *pflag.Flag) {
+		// "--skip-*" flags or other allowed flags can be set with --config
 		if isAllowedFlag(f.Name) {
-			// "--skip-*" flags or other allowed flags can be set with --config
+			// If there is any overlapping between the flag and config file, the flag will take precedence over the config file, e.g. IgnorePreflightErrors.
+			klog.Warningf("both the flag %s and config file are set", f.Name)
```

Comment:
Emm, I don't know what the key point of this is?
Can you explain more why we need it? 
This warning can be confusing to our users...

Reply:
the purpose is that letting end user knows the there is a mix-config, and potentially the config in the config file will be overwritten.

Reply:
we should just error on mixed config and flags. reset should do the same as other commands do. in the future we could start removing duplicate options between flags / config.

Reply:
Ack.

let's remove / deprecate some of the flags then.

/close